numpy/f2py/crackfortran.py

```diff
@@ -2567,11 +2590,18 @@ def get_parameters(vars, global_params={}):
                 outmess(f'get_parameters[TODO]: '
                         f'implement evaluation of complex expression {v}')
 
+            dimspec = ([s.lstrip('dimension') for s in vars[n]['attrspec']
+                       if s.startswith('dimension')] or [None])[0]
+
+            dimspec = ([s.lstrip('dimension') for s in vars[n]['attrspec']
+                       if s.startswith('dimension')] or [None])[0]
+
             try:
-                params[n] = eval(v, g_params, params)
+                params[n] = param_eval(v, g_params, params, dimspec=dimspec)
             except Exception as msg:
                 params[n] = v
-                outmess('get_parameters: got "%s" on %s\n' % (msg, repr(v)))
+                outmess('get_parameters: got "%s" on %s\n' % (msg, repr(n)))
```

Comment:
If we change `param_eval` to raise an exception this line will still show the error message and set `params[n] = v`